plugins/route/allocation/simple/plugin_test.go

```diff
@@ -39,13 +39,24 @@ func TestNewSimpleAllocationPlugin(t *testing.T) {
 			Name:             "yo!yo!@#$%%$%^&*(0){[]}:;',<>?/1.test",
 			ErrorExpectation: true,
 		},
+		{
+			Name:             "",
+			ErrorExpectation: false,
+		},
 	}
 
 	for _, tc := range tests {
-		_, err := NewSimpleAllocationPlugin(tc.Name)
+		sap, err := NewSimpleAllocationPlugin(tc.Name)
 		if err != nil && !tc.ErrorExpectation {
 			t.Errorf("Test case for %s got an error where none was expected", tc.Name)
 		}
+		if len(tc.Name) > 0 {
+			continue
+		}
+		dap := &SimpleAllocationPlugin{DNSSuffix: defaultDNSSuffix}
+		if len(tc.Name) == 0 && (sap.DNSSuffix != dap.DNSSuffix) {
```

Comment:
The len check here is unnecessary given your len check above.

Remove one of them ;-)